packages/cli/src/commands/screenshot-diff/screenshot-diff.command.ts

```diff
@@ -18,199 +24,244 @@ import {
+  headReplayId,
+  results,
+  diffOptions,
+}: {
+  baseReplayId: string;
+  headReplayId: string;
+  results: ScreenshotDiffResult[];
+  diffOptions: ScreenshotDiffOptions;
+}): void => {
+  const logger = log.getLogger(METICULOUS_LOGGER_NAME);
+
+  const missingHeadImagesResults = results.filter(
+    ({ outcome }) => outcome === "missing-head"
+  ) as ScreenshotDiffResultMissingHead[];
+  if (missingHeadImagesResults.length) {
+    const message = `Head replay is missing screenshots: ${missingHeadImagesResults
+      .map(({ baseScreenshotFile }) => baseScreenshotFile)
+      .sort()} => FAIL!`;
+    logger.info(message);
+    throw new ScreenshotDiffError(message, {
       baseReplayId,
       headReplayId,
-      stats: {
-        width: 0,
-        height: 0,
-        mismatchPixels: totalMismatchPixels,
-      },
     });
+  }
 
-    const diffUrl = await getDiffUrl(client, baseReplayId, headReplayId);
-    logger.info(`View screenshot diff at ${diffUrl}`);
-
-    comparisonResults.forEach((result) => {
-      logComparisonResultMessage(
-        logger,
-        `${Math.round(
-          result.comparisonResult.mismatchFraction * 100
-        )}% pixel mismatch for screenshot ${basename(
-          result.headScreenshotFile
-        )} (threshold is ${Math.round(diffThreshold * 100)}%)`,
-        result.outcome
-      );
-    });
+  const missingBaseImagesResults = results.filter(
+    ({ outcome }) => outcome === "missing-base"
+  ) as ScreenshotDiffResultMissingBase[];
+  if (missingHeadImagesResults.length) {
+    const message = `Notice: Base replay is missing screenshots: ${missingBaseImagesResults
+      .map(({ headScreenshotFile }) => basename(headScreenshotFile))
+      .sort()}`;
+    logger.info(message);
+  }
 
-    // Check if individual screenshot mismatch is higher than the threshold.
-    const mismatchingScreenshots = comparisonResults.filter(
-      (result) => result.outcome == "fail"
-    );
-
-    if (mismatchingScreenshots.length) {
-      logger.info(
-        `Screenshots ${mismatchingScreenshots
-          .map((result) => basename(result.headScreenshotFile))
-          .sort()} do not match!`
-      );
-      if (exitOnMismatch) {
-        process.exit(1);
-      }
-      throw new DiffError(
-        `Screenshots ${mismatchingScreenshots.map((result) =>
-          basename(result.headScreenshotFile)
-        )} do not match!`,
-        {
+  results.forEach((result) => {
+    const { outcome } = result;
+
+    if (outcome === "different-size") {
+      const message = `Screenshots ${basename(
+        result.headScreenshotFile
+      )} have different sizes => FAIL!`;
+      logger.info(message);
+      throw new ScreenshotDiffError(message, {
+        baseReplayId,
+        headReplayId,
+      });
+    }
+
+    if (outcome === "diff" || outcome === "no-diff") {
+      const mismatch = Math.round(result.mismatchFraction * 100);
+      const threshold = Math.round(diffOptions.diffThreshold * 100);
```

Comment:
[nit] I'd probably do `(result.mismatchFraction * 100).toFixed(3)` and `(diffOptions.diffThreshold * 100).toFixed(3)` since I think we'll sometimes use fraction thresholds that are < 0.01 (in current CI tests use diffThreshold of 0, and I've often seen diffs of about 0.01% of pixels, and it can be useful to know how many pixels differ -- currently [before this PR] it prints out something like "0% pixel mismatch, threshold is 0% -- FAIL" which isn't so useful).

Reply:
Done in c07f118